qt/qt_common/map_widget.cpp

```diff
@@ -337,6 +344,68 @@ void MapWidget::initializeGL()
   ASSERT(m_contextFactory == nullptr, ());
   if (!m_screenshotMode)
     m_ratio = devicePixelRatio();
+
+#if defined(OMIM_OS_MAC)
+  m_apiOpenGLES3 = false;
+#elif defined(OMIM_OS_LINUX)
+  {
+    QOpenGLFunctions * funcs = context()->functions();
+    const std::string vendor = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_VENDOR)).toStdString();
+    const std::string renderer = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_RENDERER)).toStdString();
+    const std::string version = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_VERSION)).toStdString();
+    const std::string glslVersion = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_SHADING_LANGUAGE_VERSION)).toStdString();
+    const std::string extensions = QString::fromLatin1(
+      (const char*)funcs->glGetString(GL_EXTENSIONS)).toStdString();
+
+    LOG(LINFO, ("Vendor:", vendor, "\nRenderer:", renderer,
+                "\nVersion:", version, "\nShading language version:\n", glslVersion,
+                "\nExtensions:", extensions));
+
+    if (context()->isOpenGLES())
+    {
+      LOG(LINFO, ("Context is LibGLES"));
+      // TODO: Fix the ES3 code path with ES3 compatible shader code.
+      m_apiOpenGLES3 = false;
+      constexpr const char* requiredExtensions[3] =
+        { "GL_EXT_map_buffer_range", "GL_OES_mapbuffer", "GL_OES_vertex_array_object" };
+      for (auto &requiredExtension: requiredExtensions)
+      {
+        if(context()->hasExtension(QByteArray::fromStdString(requiredExtension)))
+        {
+          LOG(LDEBUG, ("Found OpenGL ES 2.0 extension: ", requiredExtension));
+        }
+        else
+        {
+          LOG(LCRITICAL, ("A required OpenGL ES 2.0 extension is missing: ", requiredExtension));
+        }
```

Comment:
```suggestion
        if (context()->hasExtension(QByteArray::fromStdString(requiredExtension)))
          LOG(LDEBUG, ("Found OpenGL ES 2.0 extension: ", requiredExtension));
        else
          LOG(LCRITICAL, ("A required OpenGL ES 2.0 extension is missing: ", requiredExtension));
```

Reply:
The original idea was to stop execution in the `else` branch in the final version.
I have left that one out for test purposes, but anyhow changed it for now.